[17.0][MIG] loyalty_partner_applicability: Migration to 17.0

arantxa-s73 opened the pull request:
Module migrated to version 17.0

pilarvargas-tecnativa committed:
[ADD] loyalty_partner_applicability: New module

TT45413

Commit:
[UPD] Update loyalty_partner_applicability.pot

OCA-git-bot committed:
[BOT] post-merge updates

pilarvargas-tecnativa committed:
[IMP] loyalty_partner_applicability: Allow to share coupons inside same entity

There's a new configuration option in the settings to allow sharing
coupons between members of the same trading entity.

It has to be in the base module for being shared between all the rest of
applications, but each *_applicability module will contain extra bits of
code to use it.

TT45413

OCA-git-bot committed:
[BOT] post-merge updates

pilarvargas-tecnativa committed:
[FIX] loyalty_partner_applicability: add res_config_settings to init models

TT45413

Commit:
[UPD] Update loyalty_partner_applicability.pot

OCA-git-bot committed:
[BOT] post-merge updates

pilarvargas-tecnativa committed:
[FIX] loyalty_partner_applicability: Set default value for partner domain rule

It is necessary to establish the default value, in this case "[ ]" to the
rule_partners_domain field, in the create method for the rules established
in the computed method "_program_type_default_values" that establishes the
default values depending on the type of program, because when creating a
new program, these rules do not have the rule_partners_domain field defined,
as it is not contemplated, its value will be "False", which cannot be
interpreted as a correct domain. To avoid this error, the check is made in
the create method and the correct value is added.

TT44344

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (5 of 5 strings)

Translation: sale-promotion-16.0/sale-promotion-16.0-loyalty_partner_applicability
Translate-URL: https://translation.odoo-community.org/projects/sale-promotion-16-0/sale-promotion-16-0-loyalty_partner_applicability/it/

OCA-git-bot committed:
[BOT] post-merge updates

rousseldenis commented:
/ocabot migration loyalty_partner_applicability

rousseldenis commented:
@arantxa-s73 It seems you miss migration commit (where at least version change should appear in) :

![image](https://github.com/user-attachments/assets/3ebfa116-6948-4732-b0d2-687ec581e209)

arantxa-s73 committed:
[IMP] loyalty_partner_applicability: pre-commit auto fixes

arantxa-s73 committed:
[MIG] loyalty_partner_applicability: Migration to 17.0

pilarvargas-tecnativa commented on loyalty_partner_applicability/__manifest__.py:
```diff
@@ -2,7 +2,8 @@
+    "summary": "Enables the definition of a customer filter for promotion "
+    "rules that will "
     "only be applied to customers who meet the specified conditions in the filter.",
     "version": "16.0.2.0.2",
```
The version must be 17

arantxa-s73 replied:
Is modified in the following commit: [[MIG] loyalty_partner_applicability: Migration to 17.0](https://github.com/OCA/sale-promotion/pull/225/commits/389f92b663575bcd4f2cee2c64ac22edf0bb6b26)

pilarvargas-tecnativa replied:
Right, I was confused about commit, sorry

Can you push again to be able to run runbot? 

Thanks!

lmignon committed:
[IMP] loyalty_partner_applicability: Fix partner applicability

Before this commit, the partner applicability was not working according
to the addon description. For programs with multiple rules, rules defined
for specific partners were being applied to any partner as long as the
partner had a rule defined for them. The same applied to the coupon, where
a coupon defined for a specific partner could be used by any partner as long
the partner had a ruele defined for them in the same program where
the coupon was defined.

This commit fixes the issue by ensuring that the partner applicability
is only checked for the partner that the rule or coupon is defined for.
It's done by removing the rules not applicable to the the partner linked
to the sale order when the system is requesting the rules to be applied
to the products in the sale order. When a coupon is applied we also checke
if the partner is linked to the rule at the origin of the coupon. Or to
a rule linked to the loyalty card program.

lmignon committed:
[IMP] loyalty_partner_applicability: Improve the applicability of loyalty programs to partners

This change introduces a new mixin to allow the definition of the applicability
of a loyalty program to a partner. This mixin is used in the loyalty program model
to define the applicability of the program to partners. The mixin is also used
in the loyalty rule model to define the applicability of the rule to partners.

With this change, the applicability of a loyalty program to a partner can be
now defined at the program and rule level.

lmignon reviewed (approved):
LGTM (code review only with full diff with the 16.0 branch)

lmignon commented:
@arantxa-s73 Can you take a look at #265? It improves a lot the way we manage applicable partner on loyalty programs.

AndrzejGerasimukARCHIMEDES commented:
hello.
Can You tell me is there any possibility to finish this module for R17?

arantxa-s73 commented:
@lmignon brought

eugenios73 reviewed (approved):
functional review OK

eugenios73 commented:
@pilarvargas-tecnativa @rousseldenis can you review and check is ready to merge? Please

pilarvargas-tecnativa reviewed (changes requested):
The refactoring included in this migration is still pending approval and merging. This migration should not be merged with these changes, as the migration scripts are mainly missing and many things could break if this PR is merged without first merging the changes from that refactoring. Therefore, this PR should remain pending:
- [ ] https://github.com/OCA/sale-promotion/pull/265

pilarvargas-tecnativa commented:
Please review the changes made in this PR, which has already been merged.
https://github.com/OCA/sale-promotion/pull/265

pilarvargas-tecnativa commented:
Please review the changes made in this PR, which has already been merged.
https://github.com/OCA/sale-promotion/pull/265

eugenios73 committed:
[MIG] Add improves from PR 265: Migration to 17.0

eugenios73 commented:
> Please review the changes made in this PR, which has already been merged. #265

@pilarvargas-tecnativa I added the improvements from https://github.com/OCA/sale-promotion/pull/265. Can you review, please?

FYI @lmignon

pilarvargas-tecnativa commented:
> @pilarvargas-tecnativa I added the improvements from #265. Can you review, please?
> 
> FYI @lmignon

Hello @eugenios73,

I'm afraid the process of incorporating the changes has not been done correctly. First, you must bring the commits merged into 16.0 and place them before the migration commits. Then migrate as normal and incorporate the changes into the corresponding migration commit to respect the history and authorship of the commits.

arantxa-s73 commented:
superseeded by https://github.com/OCA/sale-promotion/pull/307
